src/cryptonote_core/tx_pool.cpp

```diff
@@ -900,6 +897,50 @@ namespace cryptonote
     }, false, category);
   }
   //------------------------------------------------------------------
+  bool tx_memory_pool::get_incremental_pool_info(time_t start_time, bool include_sensitive, std::vector<tx_details>& added_txs, std::vector<crypto::hash>& removed_txs) const
+  {
+    CRITICAL_REGION_LOCAL(m_transactions_lock);
+    if ((m_added_txs_start_time != (time_t)0) && (m_removed_txs_start_time != (time_t)0))
+    {
+      if ((start_time <= m_added_txs_start_time) || (start_time <= m_removed_txs_start_time))
+      {
+        // If either of the two lists do not go back far enough it's not possible to
+        // deliver incremental pool info
+        return false;
+      }
+      // The check uses "<=": We cannot be sure to have ALL txs exactly at start_time, only AFTER that time
+    }
+    else
+    {
+      return false;
+    }
+    added_txs.clear();
+    std::multimap<time_t, crypto::hash>::const_iterator pit = m_added_txs_by_time.lower_bound(start_time);
```

Comment:
Is there a specific reason why multimap is used over unordered multimap to store the txs? I thought unordered multimaps were faster and easier to build around?

Reply:
Well, ordering the hashes by time is the whole purpose of those maps. Unordered maps might be faster, but would not do the job ...